test/SampleWebAPIIntegration/FunctionalTest/RequestTelemetryTests.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace SampleWebAppIntegration.FunctionalTest
+{
+    using FunctionalTestUtils.Tests;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using System.Linq;
+    using Xunit;
+
+    public class RequestTelemetryTests : RequestTelemetryTestsBase
```

Comment:
So, we now have two functional test classes for request tracking - one for [web api](https://github.com/Microsoft/ApplicationInsights-aspnetv5/blob/develop/test/WebApiShimFw46.FunctionalTests/FunctionalTest/RequestTelemetryTests.cs) and one for [Mvc](https://github.com/Microsoft/ApplicationInsights-aspnetv5/blob/develop/test/Mvc6Framework45.FunctionalTests/FunctionalTest/RequestTelemetryTests.cs) and *no* unit tests for the request tracking middleware. How would you evaluate this implementation against the [four rules of simple design](http://martinfowler.com/bliki/BeckDesignRules.html)?

cc: @SergKanz, @abaranch, @upendras, @Sergei-Nikitin